spec/src/main/asciidoc/timeout.asciidoc

```diff
@@ -42,7 +42,16 @@ The above code-snippet means the method serviceA applies the `Timeout` policy,
 which is to fail the execution if the execution takes more than 400ms to complete even if it successfully returns.
 
 When a timeout occurs, A `TimeoutException` must be thrown.
-The `Timeout` annotation can be used together with `Fallback`, `CircuitBreaker`, `Asynchronous`, `Bulkhead` and `Retry`.
-A `Fallback` can be specified and it will be invoked if the `TimeoutException` is thrown.
-If `Timeout` is used together with `Retry`, the `TimoutException` will trigger the retry.
-When `Timeout` is used with `CircuitBreaker` and if a `TimeoutException` occurs, the failure will contribute towards the circuit open.
+The `@Timeout` annotation can be used together with `@Fallback`, `@CircuitBreaker`, `@Asynchronous`, `@Bulkhead` and `@Retry`.
+
+When `@Timeout` is used without `@Asynchronous`, on reaching the specified timeout duration, attempt will be made to interrupt the operation thread. The interruption will only work in certain scenarios. The interruption will not work for the following situations:
+
+* The thread is blocked on blocking I/O (database, file read/write), an exception is thrown only in case of waiting for a NIO channel
+* The thread isn't waiting (CPU intensive task) and isn't checking for being interrupted
+* The thread will catch the interrupted exception (with a general catch block) and will just continue processing, ignoring the interrupt
```

Comment:
I would delete the comment "(with a general catch block)" - it's just for illustration and unnecessary in a formal document

Reply:
done